esque/validation/validators.py

```diff
@@ -0,0 +1,71 @@
+import re
+from yamale.validators import Boolean, DefaultValidators, Integer, Number, Validator
+from yamale.util import isstr
+
+
+def all_validators() -> dict:
+    validators = DefaultValidators.copy()
+    validators[StringBool.tag] = StringBool
+    validators[StringInt.tag] = StringInt
+    validators[StringFloat.tag] = StringFloat
+    validators[StringDictionary.tag] = StringDictionary
+
+    return validators
+
+
+def catch_value_errors(validate):
+    """method decorator to catch ValueErrors for casts and return an error"""
+
+    def wrapper(self, value):
+        try:
+            return validate(self, value)
+        except ValueError:
+            return [f"'{value}' could not be casted to {self.tag[2:]}"]
+
+    return wrapper
+
+
+class StringBool(Boolean):
+    tag = "s_bool"
+
+    @catch_value_errors
+    def validate(self, value):
+        value = str(value)
+        if value.lower() not in ["false", "true"]:
+            raise ValueError
+        return super().validate(bool(value))
+
+
+class StringInt(Integer):
+    tag = "s_int"
+
+    @catch_value_errors
+    def validate(self, value):
+        return super().validate(int(value))
+
+
+class StringFloat(Number):
+    tag = "s_float"
+
+    @catch_value_errors
+    def validate(self, value):
+        return super().validate(float(value))
+
+
+class StringDictionary(Validator):
```

Comment:
That's actually a special case of a "dictionary". Therefore I'd rather choose a different name.
Or use regex for it. Btw. `'*'` is also a valid value for the `throttled.replicas` fields.
The regex might be: https://regex101.com/r/Phaz0a/1

Reply:
Even though I'm a big fan of regex, I'd always go for more maintainable solutions like I think I did.
Renaming and adding * makes sense though.

Reply:
Well, you _do_ use regex in your validator :stuck_out_tongue:. Also `KeyValuePair` still suggests something more general than it actually is. Only pairs of numbers are allowed.  
How about calling it `ReplicaList`?

Btw, did you know you can do:
```yaml
config:
  follower.replication.throttled.replicas: include('replica_list')
  leader.replication.throttled.replicas: include('replica_list')
---
replica_list: regex('^(?:\*|\d+:\d+(?:,\d+:\d+)*|)$')
```
Not suggesting you have to, I'm fine with using your validator as long as it's name doesn't suggest something different from what it is :)